CMakeLists.txt

```diff
@@ -1539,7 +1539,7 @@ if(BUILD_TESTS)
   # reduced single-core performance.
   exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
   string(REGEX MATCH "^.*(Xeon Phi).*$" CPU_MODEL_PHI ${CPUINFO})
-  if(NOT "${CPU_MODEL_PHI}" STREQUAL "")
+  if(NOT "${CPU_MODEL_PHI}" STREQUAL "" OR asan)
```

Comment:
I wasn't seeing this kind of issue running with an ASan build. Would you mind checking if you're still getting timeouts without this commit and without `-fno-omit-frame-pointer`? (Sorry, I know the test suite speed is kinda painful)

Reply:
I have removed `-fno-omit-frame-pointer`, but timeouts seem unchanged:
```
1265/2625 Test  #869: conditional_breakpoint_offload-no-syscallbuf ......................   Passed  261.44 sec
2620/2625 Test #2358: x86/string_instructions_async_signals_shared-32 ...................   Passed  264.50 sec
1280/2625 Test  #868: conditional_breakpoint_offload ....................................   Passed  274.21 sec
2586/2625 Test #2180: conditional_breakpoint_offload-32 .................................   Passed  329.61 sec
2/3 Test #1217: record_replay-no-syscallbuf ......   Passed  533.47 sec
2624/2625 Test #2528: record_replay-32 ..................................................   Passed  574.54 sec
2625/2625 Test #2529: record_replay-32-no-syscallbuf ....................................   Passed  608.18 sec
2621/2625 Test #1216: record_replay .....................................................   Passed  741.66 sec
```
With these tests still failing:
```
        1203 - nested_detach-no-syscallbuf (Failed)
        2515 - nested_detach-32-no-syscallbuf (Failed)
```

So from my point of view the `-fno-omit-frame-pointer` does not change much.

Reply:
Just to avoid misunderstanding - I run the test suite also in parallel (`ctest --verbose -j$(nproc)`).
Therefore the it depends much of the related other tests that run at that time.